healthcare/api-client/v1/hl7v2/hl7v2_messages_test.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2018 Google LLC All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+import pytest
+import sys
+import uuid
+
+from gcp_devrel.testing import eventually_consistent
+
+# Add datasets for bootstrapping datasets for testing
+sys.path.append(os.path.join(os.path.dirname(__file__), '..', 'datasets')) # noqa
+import datasets
+import hl7v2_stores
+import hl7v2_messages
+
+cloud_region = 'us-central1'
+project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+service_account_json = os.environ['GOOGLE_APPLICATION_CREDENTIALS']
+
+dataset_id = 'test_dataset_{}'.format(uuid.uuid4())
+hl7v2_store_id = 'test_hl7v2_store-{}'.format(uuid.uuid4())
+hl7v2_message_file = 'resources/hl7-sample-ingest.json'
+label_key = 'PROCESSED'
+label_value = 'TRUE'
+
+
+@pytest.fixture(scope='module')
+def test_dataset():
+    dataset = datasets.create_dataset(
+        service_account_json,
+        project_id,
+        cloud_region,
+        dataset_id)
+
+    yield dataset
+
+    # Clean up
+    datasets.delete_dataset(
+        service_account_json,
+        project_id,
+        cloud_region,
+        dataset_id)
+
+
+@pytest.fixture(scope='module')
+def test_hl7v2_store():
+    hl7v2_store = hl7v2_stores.create_hl7v2_store(
+        service_account_json,
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id)
+
+    yield hl7v2_store
+
+    hl7v2_stores.delete_hl7v2_store(
+        service_account_json,
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id)
+
+
+def test_CRUD_hl7v2_message(test_dataset, test_hl7v2_store, capsys):
+    hl7v2_messages.create_hl7v2_message(
+        service_account_json,
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_file)
+
+    hl7v2_message_id = ""
+    @eventually_consistent.call
```

Comment:
FYI we released gcp-devrel-py-tools 0.0.16. It allows you to have `tries` argument in the decorator.

```python
@eventually_consistent.call(tries=5)
```

This will shorten the test time on failures.

Reply:
Do you recommend adding it here?

Reply:
It's up to you. It's mostly for convenience during local development.

Reply:
@noerog We decided to remove gcp-devrel-py-tools from this repo. Sorry about back and forth. So can you replace it with `retrying` module? I think you know how (because other code has it), but feel free to ask anything.

Also, if sample code needs retrying, the sample command line tool is also flaky. Does it make sense to have retrying logic in the sample code so that command line tool will almost always work?

Reply:
I don't think we even need to retry the hl7v2_messages_list test. It was broken when you added the decorator because the API surface changed, not because it was flaky.